skorch/callbacks/lr_scheduler.py

```diff
@@ -149,12 +149,7 @@ def on_epoch_end(self, net, **kwargs):
             if callable(self.monitor):
                 score = self.monitor(net)
             else:
-                if self.lr_scheduler_.mode == 'max':
-                    score = -np.inf
-                elif self.lr_scheduler_.mode == 'min':
-                    score = np.inf
-                else:
-                    score = net.history[-1, self.monitor]
+                score = net.history[-1, self.monitor]
```

Comment:
What do you think: We could catch the `KeyError` and augment the error message by telling the user that they may possibly need to change the order of callbacks if they reference a metric that was not yet computed?

Reply:
I think its a good idea. I updated the PR to include this nicer error message.